src/app/pages/biotools-repo-page/biotools-repo-page.component.ts

```diff
@@ -386,4 +388,57 @@ export class BiotoolsRepoPageComponent {
     const downloadinfo = note.split(';');
     return downloadinfo[3].replace('config:','');
   }
+
+  private updateIcons(biotoolsApplication: BiotoolsApplication[]): void {
+
+    for(var application of biotoolsApplication) {
+      application.icons = { cloud: false, book: false, binder:false, globe: false, window: false };
+
+      console.log("Checking if application is of type Notebook");
+      if (application.toolType.length > 0 && application.toolType.indexOf('Notebook') >= 0) {
+        application.icons.book = true;
+      } else {
+      for (const d of application.download) {
+        if (d.note != null && (d.note.includes('Jupyter'))) {
+              application.icons.book = true;
+        }
+      }
+      }
+
+      console.log("Checking if application is of type Binder");
+      for (const d of application.download) {
+        if (d.note != null && (d.note.includes('BioExcel_Binder_Application'))) {
+              application.icons.binder = true;
+        }
+      }
+
+      console.log("Checking if application is deployable in Cloud");
+      for (const d of application.download) {
+      if (d.type === 'VM image' &&
+        (d.note != null && (d.note === 'BioExcel_Embassy_VM' || d.note === 'BioExcel_Embassy_NFS_Image' ||
+            d.note === 'BioExcel_Embassy_ECP_Image' || d.note === 'BioExcel_Binder_Application' ||
+            d.note.includes('BioExcel_ECP_Application'))
+        )) {
+            application.icons.cloud = true;
+      }
+      }
+
+      console.log("Checking if application is CLI/Library/Workbench/Suite/Desktop-Application");
+      if ((application.download.length > 0) &&
+          (application.toolType.indexOf('Library') >= 0
```

Comment:
Here the condition looks slightly weird. Not sure, why downloads need to exist for a tool to be any of the types listed. I would remove the first condition: `(application.download.length > 0) &&`

Reply:
changed it, to tooltype, which will make sense